spec/task_005_spec.rb

```diff
@@ -10,71 +11,16 @@
 
     lines = Task005.parse_input(input)
 
-    expect(lines).to eq(
+    expect(lines.map(&:class)).to eq(
       [
-        { point1: [0, 9], point2: [5, 9] },
-        { point1: [8, 0], point2: [0, 8] },
-        { point1: [9, 4], point2: [3, 4] },
+        Task005::Line,
+        Task005::Line,
+        Task005::Line,
```

Comment:
I would not check the class, but the actual positions of the lines:

```ruby
expect(lines[0].point1.x).to eq(0)
expect(lines[0].first.point1.y).to eq(9)
# ...
```

You could use `have_attributes` RSpec matcher to simplify it:
https://relishapp.com/rspec/rspec-expectations/v/3-10/docs/built-in-matchers/have-attributes-matcher.

Reply:
For example, for starters:

```ruby
expect(lines[0].first.point1).to have_attributes(x: 0, y: 9)
# ...
```

Reply:
And as a next step, such RSpec matchers can be nested:

```ruby
expect(lines[0].first).to have_attributes(
  point1: have_attributes(x: 0, y: 9),
  point2: have_attributes(x: 5, y: 9),
)
# ...
```

Reply:
That's what I didn't know :)